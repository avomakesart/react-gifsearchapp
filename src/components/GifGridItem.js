import React from 'react'

export default function GifGridItem({ title, url }) {
  return (
    <div className="card animate__animated animate__fadeIn">
      <img src={url} alt="Gif pic" />
      <p>{title}</p>
    </div>
  )
}
