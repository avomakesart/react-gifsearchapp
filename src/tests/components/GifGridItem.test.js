import React from 'react'
import '@testing-library/jest-dom'
import GifGridItem from '../../components/GifGridItem'
import { shallow } from 'enzyme'

describe('Testing <GifGridItem />', () => {
  const title = 'Demo title'
  const url = 'https://bluecatencode.com'
  const wrapper = shallow(<GifGridItem title={title} url={url} />)

  test('Should show the component correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should we get a paragraph with the title', () => {
    const p = wrapper.find('p')
    expect(p.text().trim()).toBe(title)
  })

  test('should we need to have img with the correct props', () => {
    const img = wrapper.find('img')

    expect(img.prop('src')).toBe(url)
    expect(img.prop('alt')).toBe(title)
  })

  test('should we need to have the correct className', () => {
    const div = wrapper.find('div')
    const className = div.prop('className')

    expect(className.includes('card animate__animated animate__fadeIn')).toBe(
      true
    )
  })
})
