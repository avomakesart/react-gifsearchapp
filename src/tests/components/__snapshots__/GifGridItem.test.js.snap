// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifGridItem /> Should show the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Demo title"
    src="https://bluecatencode.com"
  />
  <p>
    Demo title
  </p>
</div>
`;
